src/os/bluestore/KernelDevice.cc

```diff
@@ -94,6 +94,12 @@ int KernelDevice::open(const string& p)
     derr << __func__ << " open got: " << cpp_strerror(r) << dendl;
     goto out_fail;
   }
+  r = posix_fadvise(fd_direct, 0, 0, POSIX_FADV_RANDOM);
```

Comment:
@ukernel . Have you time to review this commit? Thanks!

Reply:
I think this isn't strictly necessary because direct io won't trigger readahead, but it doesn't hurt!